client/components/app-promo/index.jsx

```diff
@@ -18,82 +20,51 @@ export default React.createClass( {
 	},
 
 	getInitialState: function() {
-		let show_promo = true;
-
-		if ( store.get( 'desktop_promo_dismissed' ) ) {
-			show_promo = false;
-		}
-
-		if ( userUtils.getLocaleSlug() !== 'en' ) {
-			show_promo = false;
-		}
-
-		if ( viewport.isMobile() ) {
-			show_promo = false;
-		}
-
-		const chromeRE = /\bCrOS\b/;
-		if ( chromeRE.test( navigator.userAgent ) ) {
-			show_promo = false;
-		}
-
-		const promo_options = [
-			{ promo_code: 'a0001', message: 'WordPress.com your way  — desktop app now available for Mac, Windows, and Linux.' },
-			{ promo_code: 'a0002', message: 'Get WordPress.com app for your desktop.' },
-			{ promo_code: 'a0003', message: 'WordPress.com app now available for desktop.' },
-			{ promo_code: 'a0005', message: 'WordPress.com at your fingertips — download app for desktop.' }
-		];
-
-		const item = promo_options[ Math.floor( Math.random() * promo_options.length ) ];
 		return {
-			promo_item: item,
-			show_promo: show_promo
+			promo_item: getRandomPromo(),
+			show_promo: true
 		};
 	},
 
 	componentDidMount: function() {
 		// record promo view event
-		if ( this.state.show_promo ) {
-			analytics.tracks.recordEvent( 'calypso_desktop_promo_view', {
-				promo_location: this.props.location,
-				promo_code: this.state.promo_item.promo_code,
-			} );
-		}
+		this.props.recordTracksEvent( 'calypso_desktop_promo_view', {
+			promo_location: this.props.location,
+			promo_code: this.state.promo_item.promo_code,
+		} );
 	},
 
 	recordClickEvent: function() {
-		analytics.tracks.recordEvent( 'calypso_desktop_promo_click', {
+		this.props.recordTracksEvent( 'calypso_desktop_promo_click', {
 			promo_location: this.props.location,
 			promo_code: this.state.promo_item.promo_code
 		} );
 	},
 
 	dismiss: function() {
 		this.setState( { show_promo: false } );
-
-		// store as dismissed
-		store.set( 'desktop_promo_dismissed', true );
+		store.set( 'desktop_promo_disabled', true );
 	},
 
 	render: function() {
 		if ( ! this.state.show_promo ) {
 			return null;
 		}
 
-		const promo_link = 'https://apps.wordpress.com/desktop/?ref=promo_' + this.props.location + '_' + this.state.promo_item.promo_code;
+		const promoLink = getPromoLink( this.props.location, this.state.promo_item.promo_code );
 		return (
 			<div className="app-promo">
 				<span tabIndex="0" className="app-promo__dismiss" onClick={ this.dismiss } >
 					<Gridicon icon="cross" size={ 24 } />
 					<span className="app-promo__screen-reader-text">
-						{ this.translate( 'Dismiss' ) }
+						{ this.props.translate( 'Dismiss' ) }
 					</span>
 				</span>
 				<a
 					onClick={ this.recordClickEvent }
 					className="app-promo__link"
 					title="Try the desktop app!"
-					href={ promo_link }
+					href={ promoLink }
```

Comment:
does this justify creating a separate variable? maybe we could inline it here, especially if we destructure `props` and `state` above, which I like to generally do to be explicit

```js
render() {
	const { location, translate } = this.props;
	const { promo_item: { promo_code } } = this.state;

	…

	href={ getPromoLink( location, promo_code ) }
```